tools/calyx-ffi-macro/src/calyx.rs

```diff
@@ -0,0 +1,51 @@
+use std::{env, path::PathBuf, rc::Rc};
+
+use proc_macro::TokenStream;
+
+use crate::{parse::CalyxFFIMacroArgs, util};
+
+pub struct CalyxComponent {
+    ctx: Rc<calyx_ir::Context>,
+    index: usize,
+}
+
+impl CalyxComponent {
+    pub fn get(&self) -> &calyx_ir::Component {
+        &self.ctx.components[self.index]
+    }
+}
+
+pub fn parse_calyx_file(
+    args: &CalyxFFIMacroArgs,
+    file: PathBuf,
+) -> Result<CalyxComponent, TokenStream> {
+    // there has to be a better way to find lib
+    let home_dir = env::var("HOME").expect("user home not set");
+    let mut lib_path = PathBuf::from(home_dir);
+    lib_path.push(".calyx");
```

Comment:
@EclecticGriffin : How does `calyx` normally find the library directory when invoked through fud2?

Reply:
You have to manually specify the path via `fud2`. The lsp client also looks in `~/.calyx` (but has an option to override it).

Reply:
Can you change your comment to add a `TODO` so that we can find this shortcoming more easily?

Reply:
can confirm you use the `-l` flag in `fud2` which by default targets the stored calyx directory

Reply:
I wonder if we should define an environment var which we could have fud2 generate when it is compiling a rust testbench. We would fall back to the current implementation if that environment var was not available.

Reply:
The nice thing about this is that we'd just be able to run `cargo test` without needing to worry about `fud2`.